lib/sshca/identity.go

```diff
@@ -0,0 +1,384 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// Package sshca specifies interfaces for SSH certificate authorities
+package sshca
+
+import (
+	"fmt"
+	"maps"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/gravitational/trace"
+	"golang.org/x/crypto/ssh"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/api/constants"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/api/types/wrappers"
+	"github.com/gravitational/teleport/api/utils/keys"
+	"github.com/gravitational/teleport/lib/services"
+)
+
+// Identity is a user identity. All identity fields map directly to an ssh certificate field.
+type Identity struct {
+	ValidAfter  uint64
+	ValidBefore uint64
```

Comment:
Is there a reason we use `uint64` and not `time.Time`?

Reply:
The original reason was because these fields are `uint64` on the ssh certificate, and a lot of the logic being ported over around these fields used to operate directly on certain fields so this was easiest...

That being said, I tried to convert these over to timestamps based on this comment since I didn't see the harm and it might be more convenient... turns out that there's a constant used a sentinel value in `ValidBefore` that overflows `int64`s, so using a timestamp here makes the identity struct incompatible with some certs.  Reverted the change and will leave as-is for that reason.

Reply:
Thank you for looking into it.